### ISSUE TITLE
How Romanization is achieved

### ISSUE THREAD

**Issue by sushruth** [MrS0m30n3/google-translate#2]
This is not an issue, I just had this question : How is the project able to romanize text? Which google tranclate call or API is being used?

Thanks.

**Comment by MrS0m30n3** [comment]
Hi, @sushruth 

What exactly do you mean **in request**?

**Comment by sushruth** [comment]
Sorry for being so vague. I meant Request as in any response to one of the translate API calls.

**Comment by MrS0m30n3** [comment]
I still don't get exactly what your question is.
* If you mean using the API to romanize text then read the docs.
* If you mean using the CLI to romanize text then read the docs.

Bottomline, read the docs.

**Comment by sushruth** [comment]
I have tried reading the docs. I am unable to find which exact API call or query you use to get romanized text of text. Can you please link me to what DOCS you are referring to ?

Thanks.

**Comment by MrS0m30n3** [comment]
[Romanization example (multiple words)](https://mrs0m30n3.github.io/google-translate/docs/examples.html#text-romanization)
[Romanization API](https://mrs0m30n3.github.io/google-translate/docs/api.html#google_translate.GoogleTranslator.romanize)

**Comment by sushruth** [comment]
Thank you for being so patient. I understand that you have implemented it in your project but my question is how is the project able to romanize text? Which google tranclate call or API is being used?

**Comment by MrS0m30n3** [comment]
Basically i'm using the same Google service you use when accessing Google translate from your browser (translate.google.com).

The GoogleTranslator sends a GET request to the Google server mimicking the browser behaviour, then Google sends back a [JSON](https://en.wikipedia.org/wiki/JSON) structure which GoogleTranslator parses to extract all the requested information (including romanization).

For more details you can view the **debug** output from the CLI script or the API calls.

To view the actual request being sent run `google-translate -m r -vv αβγ` from the command line and find the line saying **Request url**.

You can also search for the line saying **JSON data** to view the actual JSON response.

**Comment by sushruth** [comment]
Oh okay. Thank you for explaining that. Turns out our IP can get banned if we access it enough number of times. :(

**Comment by MrS0m30n3** [comment]
Yes, Google bans your IP if you send too many requests in a sort period of time.

You can use multiple proxies to overcome this issue or use the official Google translate API if you use the translate functionality too often.

**Comment by sushruth** [comment]
I suppose. Thank you for the information.
